crates/ark/Cargo.toml

```diff
@@ -41,7 +41,9 @@ notify = "6.0.0"
 once_cell = "1.17.1"
 parking_lot = "0.12.1"
 regex = "1.10.0"
-reqwest = { version = "0.11.20", features = ["json"] }
+reqwest = { version = "0.12.5", features = ["json"] }
+reqwest-retry = "0.6.1"
+reqwest-middleware = "0.3.3"
```

Comment:
Good news is that both reqwest-retry and reqwest-middleware are smart enough to set `default-features = false` for their reqwest dependency
https://github.com/TrueLayer/reqwest-middleware/blob/6cd4eb93f8567bece4526f050cd6f7c2bde7371e/reqwest-middleware/Cargo.toml#L24
https://github.com/TrueLayer/reqwest-middleware/blob/6cd4eb93f8567bece4526f050cd6f7c2bde7371e/reqwest-retry/Cargo.toml#L19

Making #487 still viable even after this